src/Microsoft.ML.TimeSeries/TimeSeriesStatic.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.TimeSeriesProcessing;
+using Microsoft.ML.StaticPipe.Runtime;
+using System.Collections.Generic;
+using static Microsoft.ML.Runtime.TimeSeriesProcessing.SequentialAnomalyDetectionTransformBase<System.Single, Microsoft.ML.Runtime.TimeSeriesProcessing.IidAnomalyDetectionBase.State>;
+using static Microsoft.ML.Runtime.TimeSeriesProcessing.SequentialAnomalyDetectionTransformBase<System.Single, Microsoft.ML.Runtime.TimeSeriesProcessing.SsaAnomalyDetectionBase.State>;
+
+namespace Microsoft.ML.StaticPipe
+{
+    /// <summary>
+    /// IidChangePoint static API extension methods.
+    /// </summary>
+    public static class IidChangePointStaticExtensions
+    {
+        private sealed class OutColumn : Vector<double>
+        {
+            public PipelineColumn Input { get; }
+
+            public OutColumn(
+                Scalar<float> input,
+                int confidence,
+                int changeHistoryLength,
+                SequentialAnomalyDetectionTransformBase<System.Single, Microsoft.ML.Runtime.TimeSeriesProcessing.IidAnomalyDetectionBase.State>.MartingaleType martingale,
+                double eps)
+                : base(new Reconciler(confidence, changeHistoryLength, martingale, eps), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly int _confidence;
+            private readonly int _changeHistoryLength;
+            private readonly SequentialAnomalyDetectionTransformBase<System.Single, Microsoft.ML.Runtime.TimeSeriesProcessing.IidAnomalyDetectionBase.State>.MartingaleType _martingale;
+            private readonly double _eps;
+
+            public Reconciler(
+                int confidence,
+                int changeHistoryLength,
+                SequentialAnomalyDetectionTransformBase<System.Single, Microsoft.ML.Runtime.TimeSeriesProcessing.IidAnomalyDetectionBase.State>.MartingaleType martingale,
+                double eps)
+            {
+                _confidence = confidence;
+                _changeHistoryLength = changeHistoryLength;
+                _martingale = martingale;
+                _eps = eps;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
```

Comment:
does this need to get used?

Reply:
no, even if it is not a required param, you still have to include it. 

---
In reply to: [236480124](https://github.com/dotnet/machinelearning/pull/1656#discussion_r236480124) [](ancestors = 236480124,236470960)